src/Core/Listener/EmailConfirmationMailer.php

```diff
@@ -11,20 +11,22 @@
 
 namespace Flarum\Core\Listener;
 
-use Flarum\Core;
 use Flarum\Core\EmailToken;
+use Flarum\Core\Jobs\MailJob;
+use Flarum\Core\Support\DispatchJobsTrait;
 use Flarum\Core\User;
 use Flarum\Event\UserEmailChangeWasRequested;
 use Flarum\Event\UserWasRegistered;
 use Flarum\Forum\UrlGenerator;
 use Flarum\Settings\SettingsRepositoryInterface;
+use Illuminate\Contracts\Bus\Dispatcher as JobDispatcher;
 use Illuminate\Contracts\Events\Dispatcher;
 use Illuminate\Contracts\Mail\Mailer;
-use Illuminate\Mail\Message;
 use Symfony\Component\Translation\TranslatorInterface;
 
 class EmailConfirmationMailer
 {
+    use DispatchJobsTrait;
```

Comment:
To be fair, that one at least encapsulates [some logic](https://github.com/flarum/core/blob/01eba18164d6eedd7cd3b58169fa5779ae6306ad/src/Core/Support/DispatchEventsTrait.php#L32-L36) ;)